beacon-chain/state/stategen/service.go

```diff
@@ -8,17 +8,38 @@ import (
 	"errors"
 	"sync"
 
+	eth "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
 	"github.com/prysmaticlabs/prysm/beacon-chain/db"
 	"github.com/prysmaticlabs/prysm/beacon-chain/state"
+	ethereum_beacon_p2p_v1 "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
 	"github.com/prysmaticlabs/prysm/shared/bytesutil"
 	"github.com/prysmaticlabs/prysm/shared/params"
 	"go.opencensus.io/trace"
 )
 
 var defaultHotStateDBInterval uint64 = 128 // slots
 
-// State represents a management object that handles the internal
+// Service represents a management object that handles the internal
 // logic of maintaining both hot and cold states in DB.
+type Service interface {
```

Comment:
This is a very generically named iface. It conflicts with the name of the service interface in our shared service registry. Can we find a more apt interface perhaps something such as StateManager?